internal/tiger/cmd/db.go

```diff
@@ -932,3 +960,363 @@ func isConnectionRejected(err error) bool {
+	result, err := common.SavePasswordWithResult(service, newPassword, role)
+	if err != nil {
+		fmt.Fprintf(statusOut, "Warning: could not save password: %v\n", err)
+	} else if result.Success {
+		fmt.Fprintf(statusOut, "%s\n", result.Message)
+	}
+
+	return nil
+}
+
+// resetServicePassword generates a new password and updates it via API
+func resetServicePassword(ctx context.Context, cfg *config.Config, service api.Service, role string, statusOut io.Writer) (string, error) {
+	// Generate new password
+	newPassword, err := generateSecurePassword(32)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate new password: %w", err)
+	}
+
+	// Get API credentials
+	apiKey, projectID, err := getCredentialsForService()
+	if err != nil {
+		return "", fmt.Errorf("authentication required: %w. Please run 'tiger auth login'", err)
+	}
+
+	// Create API client
+	client, err := api.NewTigerClient(cfg, apiKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to create API client: %w", err)
+	}
+
+	// Update and save password
+	if err := updateAndSaveServicePassword(ctx, client, projectID, service, newPassword, role, statusOut); err != nil {
+		return "", err
+	}
+
+	fmt.Fprintf(statusOut, "Password reset successfully for '%s' user\n", role)
+	return newPassword, nil
+}
+
+// connectOptions holds options for the connect command
+type connectOptions struct {
+	password      string // Explicit password to use (headless mode)
+	resetPassword bool   // Reset password before connecting (headless mode)
+}
+
+// testSaveAndLaunchPsqlWithPassword tests a password, saves it if valid, and launches psql.
+// Returns a retryable error if authentication fails, or a fatal error otherwise.
+func testSaveAndLaunchPsqlWithPassword(
+	ctx context.Context,
+	cmd *cobra.Command,
+	details *common.ConnectionDetails,
+	psqlPath string,
+	psqlFlags []string,
+	service api.Service,
+	password string,
+	role string,
+) error {
+	// Test the password
+	err := testConnectionWithPassword(ctx, details, password, 10*time.Second)
+	if err != nil {
+		return err
+	}
+
+	// Password works! Save it
+	result, saveErr := common.SavePasswordWithResult(service, password, role)
+	if saveErr != nil {
+		fmt.Fprintf(cmd.ErrOrStderr(), "Warning: could not save password: %v\n", saveErr)
+	} else if result.Success {
+		fmt.Fprintf(cmd.ErrOrStderr(), "%s\n", result.Message)
+	}
+
+	// Launch psql (password is now in storage, launchPsql will retrieve it)
+	return launchPsql(details.String(), psqlPath, psqlFlags, service, role, cmd)
+}
+
+// resetPasswordAndLaunchPsql resets the password via API and launches psql with the new password.
+func resetPasswordAndLaunchPsql(
+	ctx context.Context,
+	cmd *cobra.Command,
+	cfg *config.Config,
+	details *common.ConnectionDetails,
+	psqlPath string,
+	psqlFlags []string,
+	service api.Service,
+	role string,
+) error {
+	_, err := resetServicePassword(ctx, cfg, service, role, cmd.ErrOrStderr())
+	if err != nil {
+		return err
+	}
```

Comment:
It's slightly more idiomatic to scope `err` to the conditional block, imo:

```suggestion
	if _, err := resetServicePassword(ctx, cfg, service, role, cmd.ErrOrStderr()); err != nil {
		return err
	}
```